packages/ember-runtime/lib/mixins/container_proxy.js

```diff
@@ -55,7 +55,14 @@ export default Mixin.create({
    @param {Object} options
    @return {any}
    */
-  lookup: containerAlias('lookup'),
+  lookup() {
+    let container = get(this, '__container__');
+    let result = container.lookup(...arguments);
+    if (result && result.owner === undefined) {
+      result.owner = this;
```

Comment:
Yeah, definitely open to alternatives to `owner` - it's short and sweet, but that also means it might be commonly used.